network/gossip/libp2p/peerInfoCache.go

```diff
@@ -0,0 +1,56 @@
+package libp2p
+
+import (
+	"fmt"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+var infoCache *peerInfoCache
```

Comment:
From PR description:
> Defines a PeerAddressInfo cache to cache the flow.Identity to libp2p peer.AddressInfo conversion results.

Is this conversion expensive? It looks like it's just one hashing operation. And we would only need to do this when the identity list changes, which is infrequent on the order of hours if not days. This doesn't seem to me like something we need to cache.

Reply:
I assumed that the key conversion (flow crypto key format to libp2p key format) _may_ be expensive and hence added the cache. The conversion will happen every time `updatePeer` runs since the connector attempts to connect to all ids over and over again. Libp2p ignores connection attempt to the same id if already connected.

I did some benchmarking of the two functions that make up the bulk of this function:

> func BenchmarkPeerInfoFromID(b *testing.B) {
> 	b.StopTimer()
> 	id := unittest.IdentityFixture()
> 	key, _ := generateFlowNetworkingKey(id.NodeID)
> 	id.NetworkPubKey = key.PublicKey()
> 	id.Address = "1.1.1.1:3569"
> 	b.StartTimer()
> 	for n := 0; n < b.N; n++ {
> 		nodeAddress, _ := nodeAddressFromIdentity(*id)
> 		_, _ = GetPeerInfo(nodeAddress)
> 	}
> }
And it doesn't seem that expensive:

> BenchmarkPeerInfoFromID-8         504056              2270 ns/op            1298 B/op         32 allocs/op

Reply:
Thanks for confirming with the benchmark 👍